src/jsDelivr/pkgTypes.ts

```diff
@@ -0,0 +1,103 @@
+import type { RawPkg, StyleType } from '../@types/pkg';
+import { datadog } from '../utils/datadog';
+
+import type { File } from './index';
+
+const styleFileExtensions = ['css', 'less', 'scss'];
+const styleFilePattern = createFilePattern(styleFileExtensions);
+
+const jsFileExtensions = ['js', 'mjs', 'cjs'];
+const jsFilePattern = createFilePattern(jsFileExtensions);
+
+function createFilePattern(extensions: string[]): RegExp {
+  const extPattern = extensions.join('|');
+
+  // https://regex101.com/r/X5jQfH/2
+  return new RegExp(
+    `^(?:(?!\\/(docs?|documentation|examples?|samples?|demos?|tests?)\\/)(?!\\/[._]).)+\\.(${extPattern})$`
+  );
+}
+
+export function getStyleTypes(
+  pkg: Pick<RawPkg, 'styleTypes'>,
+  filelist: File[]
+): Pick<RawPkg, 'styleTypes'> {
+  const start = Date.now();
+
+  try {
+    const styleTypes = new Set<StyleType>(pkg.styleTypes);
+
+    for (const file of filelist) {
+      if (!styleFilePattern.test(file.name)) {
+        continue;
+      }
+
+      const type = file.name.split('.').pop();
+
+      if (type) {
+        styleTypes.add(type);
```

Comment:
This should be lowercase as well.

Reply:
Actually, it is now, because the regex isn't case-insensitive - but it should be case-insensitive and normalized to lc.

Reply:
fixed https://github.com/algolia/npm-search/pull/683/commits/8dd49c3c9ebb44191d4a82cf699bb21d4d86a0d0